frontend/src/bundles/candidate/components/candidate-modal/candidate-modal.tsx

```diff
@@ -0,0 +1,337 @@
+import {
+    Add as AddIcon,
+    DeleteOutline as DeleteIcon,
+} from '@mui/icons-material';
+import { FormHelperText } from '@mui/material';
+import {
+    Controller,
+    type ControllerRenderProps,
+    type SubmitHandler,
+    useFieldArray,
+} from 'react-hook-form';
+
+import {
+    Button,
+    Checkbox,
+    FormControl,
+    FormLabel,
+    Grid,
+    Input,
+    Modal,
+    Textarea,
+    Typography,
+} from '~/bundles/common/components/components.js';
+import { getValidClassNames } from '~/bundles/common/helpers/helpers.js';
+import {
+    useAppDispatch,
+    useAppForm,
+    useAppSelector,
+    useCallback,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
+
+import { actions as candidateActions } from '../../store/candidate.js';
+import { type ContactCandidateDto } from '../../types/types.js';
+import { ContactCandidateValidationSchema } from '../../validation-schemas/validation-schemas.js';
+import { DEFAULT_CONTACT_CANDIDATE_MODAL } from './constants.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+    isOpen: boolean;
+    onClose: () => void;
+};
+
+const CandidateModal: React.FC<Properties> = ({ isOpen = true, onClose }) => {
+    const [isSaveTemplateChecked, setIsSaveTemplateChecked] = useState(false);
+
+    const dispatch = useAppDispatch();
+    const { messageTemplates } = useAppSelector(({ candidate }) => ({
+        messageTemplates: candidate.messageTemplates,
+    }));
+
+    const { control, errors, handleSubmit, setValue } =
+        useAppForm<ContactCandidateDto>({
+            defaultValues: DEFAULT_CONTACT_CANDIDATE_MODAL,
+            validationSchema: ContactCandidateValidationSchema,
+        });
+
+    const { fields, append, remove } = useFieldArray({
+        name: 'links',
+        control,
+    });
+
+    const addLink = useCallback(() => {
+        append({
+            value: '',
+        });
+    }, [append]);
+
+    const removeLink = useCallback(
+        (index: number) => (): void => {
+            remove(index);
+        },
+        [remove],
+    );
+
+    const applyTemplate = useCallback(
+        (message: string) => (): void => {
+            setValue('message', message);
+        },
+        [setValue],
+    );
```

Comment:
Don't use useCallback in every function (https://dmitripavlutin.com/react-usecallback/)